### PR TITLE
use gopath instead of home in TestBuildDataDirDotOk

### PR DESCRIPTION
**Pull request by therealssj**
Changes:
-
Check for `GOPATH` prefix instead of `~` in `TestBuildDataDirDotOk` as when tests are run `go test` goes to the directory of the file containing the test and creates all files/dirs relative to that. Since go projects are under the `GOPATH` directory its better to check for that.

Does this change need to mentioned in CHANGELOG.md?
No

### CODE REVIEW

File: src/util/file/file_test.go [hunk 1]
```diff
@@ -61,12 +61,12 @@ func TestBuildDataDirDotOk(t *testing.T) {
 	cleanDir := filepath.Clean(dir)
 	require.True(t, strings.HasSuffix(builtDir, cleanDir))
 
-	home := filepath.Clean(UserHome())
-	if home == "" {
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
```
Comment (iketheadore) [inline src/util/file/file_test.go]: When `GOPATH` is not set, Go will use `$HOME/go` as `GOPATH`, so the test will fail here.

**Comment by gz-c** [comment]
@iketheadore @therealssj what is the fix?

**Comment by iketheadore** [comment]
set the gopath variable to $home/go if it's empty.

**Comment by therealssj** [comment]
done

File: src/util/file/file_test.go [hunk 2]
```diff
@@ -62,12 +62,14 @@ func TestBuildDataDirDotOk(t *testing.T) {
...
 
 	gopath := os.Getenv("GOPATH")
+	// by default go uses GOPATH=$HOME/go if it is not set
 	if gopath == "" {
-		require.Equal(t, cleanDir, builtDir)
-	} else {
-		require.True(t, strings.HasPrefix(builtDir, gopath))
-		require.NotEqual(t, builtDir, filepath.Clean(gopath))
+		home := filepath.Clean(UserHome())
+		gopath = home + "/" + "go"
```
Comment (iketheadore) [inline src/util/file/file_test.go]: It's better to use `filepath.Join(home, "go") to be able to compatible with different systems.

Comment (therealssj) [reply src/util/file/file_test.go]: Oh right! Thnx!

**Pull request merged**